pydantic_ai_slim/pydantic_ai/result.py

```diff
@@ -48,30 +51,49 @@
 Usage `ResultValidatorFunc[AgentDepsT, T]`.
 """
 
+DEFAULT_OUTPUT_TOOL_NAME = 'final_output'
+
 
 @dataclass
-class AgentStream(Generic[AgentDepsT, ResultDataT]):
+class ToolStructuredOutput(Generic[OutputDataT]):
```

Comment:
@dmontagu please look at this and check if you're happy, it needs more tests and docs, but I think the principle should be clear.

Reply:
I don't love this name, partially because it's kind of wordy, partially because I want a name for a function that you call and use its result as the final output, and I feel like whatever that thing gets called will be difficult to distinguish from this name. In particular, I wanted to use `OutputTool` for that, but it seems confusing for `OutputTool` and `ToolStructuredOutput` to be different things.

I'll try to think of a better name for this ASAP..

Reply:
Actually, I feel like we could just use `OutputTool` as long as it's compatible with the "structured output" version _and_ the (python) function calling version. It makes sense, considering that for many/most types you'd use with `ToolStructuredOutput`, it's basically doing a function call with the model's response, just to the type's initializer instead of an arbitrary callable.

So I'd propose we rename this to `OutputTool`, and plan to (eventually) support instantiating it with arbitrary callables (not just types)

Reply:
I did the renaming, and tweaked the init signature to be a bit more future-proof (in particular, I've made it keyword-only). I'd be happy to change that in the future, just don't want that to break things when we add support for callables if getting type-checking to work requires some tweaks.

I made the keyword argument's keyword `type_`, but I'm open to changing it to `type` or `tp` or whatever you think makes sense. (`output_type` seems unfortunate given it's already an `OutputTool` but could do that too).

Reply:
my only concern is that `OutputTool` feels like a kind of tool not a kind of structured output. It's not obvious what we would call "output using JSON Schema prompt" as an alternative to `OutputTool`? `OutputJSONSchema` feels confusing.

Could we call it `OutputWithTool`, then in future add `OutputWithJSONSchema`?

See [instructor `Mode`](https://github.com/instructor-ai/instructor/blob/611d7a6d119d2493d5f24a8e484aeb3c51a9c9e4/instructor/mode.py#L9) for reference.

Reply:
Isn't it a kind of tool though?

Reply:
Okay thinking about it more, I'd be fine with `OutputContent` or `OutputJson` for the case you are describing.

I'd also be okay with `ToolOutput` and `PromptSchemaOutput` and `StructuredOutput`, or similar.

But I'd rather not include `With` in the name if _at all_ possible, that just seems unnecessarily verbose.

I have a weak preference for the name _starting_ with `Output` instead of ending for discoverability, but it's not the end of the world; if it ends with `Output` though we should maybe put them all in a dedicated module to help with discoverability in that way.

Reply:
I'd also be okay just having a single `Output` type with different `__init__` signatures for the different modes. I see pros and cons to that though; I'm still fine having separate types for the separate modes.

Reply:
okay, I renamed OutputTool to ToolOutput and I think that's good enough for this